mailstore-connector/src/com/axonivy/connector/oauth/ssl/SSLContextConfigure.java

```diff
@@ -1,88 +0,0 @@
-
-import ch.ivyteam.ivy.ssl.restricted.SslClientSettings;
-
-public class SSLContextConfigure {
-	private static final SSLContextConfigure INSTANCE = new SSLContextConfigure();
-	private SSLContextConfigure() {}
-	
-	public static SSLContextConfigure get() {
-		return INSTANCE;
-	}
-
-	public boolean isStartTLSEnabled(Properties properties) {
-		if (properties == null) {
-			return false;
-		}
-		return BooleanUtils.toBoolean(properties.getProperty(StartTLS.ENABLE.getProperty()))
-				|| BooleanUtils.toBoolean(properties.getProperty(StartTLS.REQUIRED.getProperty()));
-	}
-
-	public void addIvyTrustStoreToCurrentContext() throws NoSuchAlgorithmException, KeyStoreException,
-			CertificateException, IOException, KeyManagementException {
-		TrustManagerFactory tmFactory = initDefaultTrustManagerFactory();
-		// Backup default Certificates
-		X509TrustManager defaultX509CertTM = getFirstX509TrustManagerFromFactory(tmFactory);
-		var ivyTrustStore = SslClientSettings.instance().getTrustStore();
-		try (InputStream trustStoreStream = Files.newInputStream(ivyTrustStore.getPath())) {
-			KeyStore ivyKeyStore = KeyStore.getInstance(ivyTrustStore.getType());
-			ivyKeyStore.load(trustStoreStream, ivyTrustStore.getPassword());
-			tmFactory = getDefaultAlgorithm();
-			tmFactory.init(ivyKeyStore);
-			X509TrustManager ivyTrustManager = getFirstX509TrustManagerFromFactory(tmFactory);
-			// Merge ivyTrustStrore with defaultTrustStore then build SSLContext
-			buildSSLContext(new MailStoreTrustManager(defaultX509CertTM, ivyTrustManager));
-		}
-	}
-
-	private TrustManagerFactory getDefaultAlgorithm() throws NoSuchAlgorithmException {
-		return TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-	}
-
-	private X509TrustManager getFirstX509TrustManagerFromFactory(TrustManagerFactory trustManagerFactory) {
-		return Stream.of(trustManagerFactory.getTrustManagers())
-				.filter(trustManager -> trustManager instanceof X509TrustManager)
-				.findFirst().map(trustManager -> (X509TrustManager) trustManager)
-				.orElse(null);
-	}
-
-	private TrustManagerFactory initDefaultTrustManagerFactory() throws NoSuchAlgorithmException, KeyStoreException {
-		TrustManagerFactory trustManagerFactory = getDefaultAlgorithm();
-		trustManagerFactory.init((KeyStore) null);
-		return trustManagerFactory;
-	}
-
-	public void resetToDefaultTrustStore() throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
-		TrustManagerFactory trustManagerFactory = initDefaultTrustManagerFactory();
-		X509TrustManager defaultX509CertTM = getFirstX509TrustManagerFromFactory(trustManagerFactory);
-		buildSSLContext(defaultX509CertTM);
-	}
-
-	private void buildSSLContext(X509TrustManager x509CertTrustManager)
-			throws NoSuchAlgorithmException, KeyManagementException {
-		SSLContext sslContext = SSLContext.getInstance(Constants.TLS);
-		sslContext.init(null, new TrustManager[] { x509CertTrustManager }, null);
-		SSLContext.setDefault(sslContext);
```

Comment:
this is the main thing to be removed: now we do no longer "pollute" the global SSLContext.